packages/database/src/migrations/20210922211939-LESMIS-number-of-schools-overlays-modifies-data.js

```diff
@@ -0,0 +1,324 @@
+'use strict';
+
+import { generateId, insertObject, codeToId, nameToId, arrayToDbString } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const getReport = (reportCode, dataElements) => {
+  const aggregations = [
+    {
+      type: 'FINAL_EACH_YEAR',
+      config: {
+        dataSourceEntityType: 'sub_district',
+        aggregationEntityType: 'requested',
+      },
+    },
+  ];
+
+  if (reportCode.includes('Province')) {
+    aggregations.push({
+      type: 'SUM_PER_ORG_GROUP',
+      config: {
+        dataSourceEntityType: 'sub_district',
+        aggregationEntityType: 'district',
+      },
+    });
+  }
+
+  return {
+    code: reportCode,
+    config: {
+      fetch: {
+        dataElements,
+        aggregations,
+      },
+      transform: [
+        {
+          transform: 'insertColumns',
+          columns: {
+            '=$dataElement': '=$value',
+          },
+        },
+        {
+          transform: 'updateColumns',
+          insert: {
+            organisationUnitCode: '=$organisationUnit',
+            value: `=sum(${dataElements.map(code => `$${code}`)})`,
+          },
+          include: ['period'],
+        },
+        {
+          transform: 'mergeRows',
+          groupBy: ['organisationUnitCode', 'period'],
+          using: 'sum',
+        },
+      ],
```

Comment:
```suggestion
      transform: [
        {
          transform: 'updateColumns',
          insert: {
            organisationUnitCode: '=$organisationUnit',
          },
          include: ['period', 'value'],
        },
        {
          transform: 'mergeRows',
          groupBy: ['organisationUnitCode', 'period'],
          using: 'sum',
        },
      ],
```

You wind up converting the value back and forth. You convert the values to named fields based on the `dataElement`.

But then you sum all the `dataElements` when each row only has *one* of those fields with a value (you haven't merged any rows at that point). So all you wind up doing is summing the one `dataElement` you have, and a whole bunch of `undefined`, which means you just convert that named field back to being named `value`

Reply:
Makes sense. This is much better thanks.